src/main/java/org/opensearch/index/codec/customcodecs/Lucene99CustomStoredFieldsFormat.java

```diff
@@ -17,15 +17,17 @@
 import org.apache.lucene.index.SegmentInfo;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.IOContext;
+import org.opensearch.index.codec.customcodecs.backward_codecs.Lucene95CustomCodec;
 
 import java.io.IOException;
 import java.util.Objects;
 
 /** Stored field format used by pluggable codec */
-public class Lucene95CustomStoredFieldsFormat extends StoredFieldsFormat {
+public class Lucene99CustomStoredFieldsFormat extends StoredFieldsFormat {
 
     /** A key that we use to map to a mode */
-    public static final String MODE_KEY = Lucene95CustomStoredFieldsFormat.class.getSimpleName() + ".mode";
+    public static final String LUCENE95_MODE_KEY = "Lucene95CustomStoredFieldsFormat.mode";
```

Comment:
We would need the ```LUCENE95_MODE_KEY``` to read the old segments I believe. We may not need a new CustomStoredFieldsFormat because Lucene95 and Lucene99 both use ```Lucene90StoredFieldsFormat```.
We should be able to read the segments with the old mode key. wdyt?

Reply:
The old segments will be read by old codec, with  the `Lucene95CustomStoredFieldsFormat`

Reply:
Yes, mode is different. If you look at the overridden fieldsReader method, we are first checking if its an old segment (Lucene95CustomMode), if yes, then accly we read the mode, and pass it on to the impl method.
There are two impl methods currently to handle both the types of modes.

Reply:
But that's the problem - the old segments will be read by old codec only, and new by new codec - the new codec should never see old segments and should not expect them (same for old codec)

Reply:
got it, we can keep two storedfieldsformat in that case. I'll work on that change.